### DISCUSSION TITLE
Some album covers are empty

### DISCUSSION THREAD

**Discussion by Cartoon-Dude** [LycheeOrg/Lychee#2658] [Bugs]
### Lychee version

6.0.1

### Which PHP version are you using?

PHP 8.3

### Detailed description of the problem

After updating there's some random albums not having any cover preview.
![image](https://github.com/user-attachments/assets/969cd22a-571e-4025-854b-072a63a7fdf4)
Having this in the errors.log
![image](https://github.com/user-attachments/assets/d2d3135f-8164-4072-bbbb-1c3e2216130d)

### Steps to reproduce the issue

Updating to 6.0.1

### Diagnostics [REQUIRED]

Info

Lychee Version (git):                      master (08e06f8) -- Up to date (-0.0014311699959411 years ago).
DB Version:                                6.0.1
composer install:                          --no-dev
APP_ENV:                                   production
APP_DEBUG:                                 false
APP_URL:                                   set
APP_DIR:                                   default
LOG_VIEWER_ENABLED:                        true
VUEJS_ENABLED:                             true
PHOTO_PIPES:                               true
System:                                    Linux
PHP Version:                               8.3.6
PHP User agent:                            Lychee/4 (https://lycheeorg.github.io/)
Timezone:                                  Europe/Zurich
Max uploaded file size:                    256M
Max post size:                             256M
Chunk size:                                85.00 MB
Max execution time:                        200
MySQL Version:                             10.11.6-MariaDB
exec() Available:                          yes
Imagick Available:                         1
Imagick Enabled:                           1
Imagick Version:                           1692
GD Version:                                2.3.3
Number of foreign key:                     12 found.

Configuration

version:                                   060001
check_for_updates:                         1
sorting_photos_col:                        taken_at
sorting_photos_order:                      ASC
sorting_albums_col:                        created_at
sorting_albums_order:                      DESC
imagick:                                   1
skip_duplicates:                           0
small_max_width:                           0
small_max_height:                          360
medium_max_width:                          1920
medium_max_height:                         1080
lang:                                      fr
image_overlay_type:                        desc
default_license:                           CC-BY-NC-ND-4.0
compression_quality:                       90
grants_full_photo_access:                  1
delete_imported:                           0
mod_frame_enabled:                         1
mod_frame_refresh:                         30
thumb_2x:                                  1
small_2x:                                  1
medium_2x:                                 1
landing_page_enable:                       0
force_migration_in_production:             0
search_public:                             0
hide_version_number:                       0
SL_enable:                                 0
SL_for_admin:                              0
recent_age:                                1
grants_download:                           0
photos_wraparound:                         1
map_display:                               0
zip64:                                     1
map_display_public:                        0
map_provider:                              Wikimedia
force_32bit_ids:                           0
map_include_subalbums:                     0
update_check_every_days:                   3
has_exiftool:                              0
share_button_visible:                      0
import_via_symlink:                        0
has_ffmpeg:                                0
location_decoding:                         0
location_decoding_timeout:                 30
location_show:                             0
location_show_public:                      0
rss_enable:                                0
rss_recent_days:                           7
rss_max_items:                             100
prefer_available_xmp_metadata:             0
editor_enabled:                            1
lossless_optimization:                     0
swipe_tolerance_x:                         150
swipe_tolerance_y:                         250
local_takestamp_video_formats:             .avi|.mov
log_max_num_line:                          1000
unlock_password_photos_with_url_param:     0
nsfw_visible:                              1
nsfw_blur:                                 0
nsfw_warning:                              0
nsfw_warning_admin:                        0
nsfw_banner_override:                      
map_display_direction:                     1
album_subtitle_type:                       oldstyle
upload_processing_limit:                   4
new_photos_notification:                   0
legacy_id_redirection:                     1
zip_deflate_level:                         6
default_album_protection:                  1
album_decoration:                          layers
album_decoration_orientation:              row
allow_username_change:                     1
auto_fix_orientation:                      1
use_job_queues:                            0
random_album_id:                           starred
use_last_modified_date_when_no_exif_date:  0
layout:                                    justified
date_format_photo_thumb:                   M j, Y, g:i:s A e
date_format_photo_overlay:                 M j, Y, g:i:s A e
date_format_sidebar_uploaded:              M j, Y, g:i:s A e
date_format_sidebar_taken_at:              M j, Y, g:i:s A e
date_format_hero_min_max:                  F Y
date_format_hero_created_at:               M j, Y, g:i:s A T
date_format_album_thumb:                   M Y
upload_chunk_size:                         0
nsfw_banner_blur_backdrop:                 0
search_pagination_limit:                   1000
search_minimum_length_required:            4
photo_layout_justified_row_height:         320
photo_layout_masonry_column_width:         300
photo_layout_grid_column_width:            250
photo_layout_square_column_width:          200
photo_layout_gap:                          12
display_thumb_album_overlay:               always
display_thumb_photo_overlay:               hover
default_album_thumb_aspect_ratio:          1/1
use_album_compact_header:                  0
login_button_position:                     left
back_button_enabled:                       0
back_button_text:                          Return to Home
current_job_processing_visible:            1
SA_random_thumbs:                          0
job_processing_queue_visible:              1
login_required:                            0
enable_unsorted:                           1
enable_starred:                            1
enable_recent:                             1
enable_on_this_day:                        1
login_required_root_only:                  1
dark_mode_enabled:                         1
show_keybinding_help_popup:                1
show_keybinding_help_button:               1
disable_se_call_for_actions:               0
enable_se_preview:                         0
slideshow_timeout:                         5
keep_original_untouched:                   1
hide_nsfw_in_smart_albums_and_search:      1

### Browser & System [REQUIRED]

Firefox 132

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by ildyria** [comment]
That is very very weird.
Also it is as if the albums are empty. Are they password protected by any chance?
Do you have a view when you are logged in?


The issue in the logs is unrelated to this.

**Reply by Cartoon-Dude** [reply]
No, they're in majority not password protected, and same when logged in.
When clicking an album there's all the pictures inside as normally.

**Reply by ildyria** [reply]
Can you go check the maintenance page, I suspect you have smalls that need to be generated. :)

**Reply by Cartoon-Dude** [reply]
Indeed there was ~3400 needing to generate.
Used the script to do this, but still the same problem.

**Reply by ildyria** [reply]
That is surprising. >_<

**Reply by Cartoon-Dude** [reply]
Deleting all the photos of one album and reuploading them: same problem.
Creating a new album with the same pictures: also same problem.
Also having some weird artefacts after uploading, but when refreshing it was fine.
![image](https://github.com/user-attachments/assets/cb932cf9-84db-4ab5-9515-208f44d0a837)

**Reply by Cartoon-Dude** [reply]
Huh, now having:
info Found 3 small that could be generated.
info Found 7 small2x that could be generated.
info Found 8 medium that could be generated.
info Found 11 placeholder images that have not been encoded.
info Found 5333 placeholders that could be generated.
What is the command to generate placeholders?

**Reply by ildyria** [reply]
> Also having some weird artefacts after uploading, but when refreshing it was fine.

I noticed that once since I added the placeholder in the albums photos and not just to the album thumbs.

You can generate the place holder either in the maintenance page or via command:
```
php artisan lychee:generate_thumbs --help
```

**Reply by Cartoon-Dude** [reply]
Thank you, it is now all generated.
Just remaining "Found 5341 placeholder images that have not been encoded."
How to encode them?

**Reply by ildyria** [reply]
```
php artisan lychee:encode_placeholders --help
```

**Reply by ildyria** [reply]
Still that will not solve the issue. I am unsure what is going wrong in your case.
